pkg/kubernetes/details.js

```diff
@@ -3,12 +3,119 @@ define([
     "base1/cockpit",
     "base1/angular",
     "base1/term",
+    "kubernetes/client",
     "kubernetes/app"
-], function($, cockpit, angular, Terminal) {
+], function($, cockpit, angular, Terminal, kubernetes) {
     'use strict';
 
     var _ = cockpit.gettext;
 
+    var k8client =  kubernetes.k8client();
+    var adjust_btn = $('#adjust-entity-button'); 
+    var adjust_entity_dlg = $('#adjust-entity-dialog');
+    var delete_btn = $('#delete-entity-button'); 
+    var delete_entity_dlg = $('#delete-entity-dialog');
+    var delete_pod_dlg = $('#delete-pod-dialog');
+    var delete_pod_btn = $('#delete-pod-button');
+
+    delete_entity_dlg.on('show.bs.modal', function(e) {
+        var key = $(e.relatedTarget).attr("data-key");
+        var entity = k8client.objects[key];
+        delete_entity_dlg.find('.modal-body').text(cockpit.format(_("Delete $0 $1?"), entity.kind, entity.metadata.name));
+
+        delete_btn.on('click', function() {
+            k8client.remove(entity.metadata.selfLink);
```

Comment:
Please attach the promise to the dialog so that a spinner is shown as you would expect.